data_structure/ordered_set/gen/max_random.cpp

```diff
@@ -0,0 +1,117 @@
+#include <iostream>
+#include <string>
+#include <set>
+#include <vector>
+#include <cstdlib>
+#include <cstdio>
+
+#include "../params.h"
+#include "random.h"
+
+using namespace std;
+
+/*
+要素の種類 K
+0:1000, dense
+1:1000, random
+2:10^6, dense
+3:10^6, random
+
+クエリ
+0:[1/6,1/6,1/6,1/6,1/6,1/6]
+1:[1/5,eps,1/5,1/5,1/5,1/5]
+2:[1,eps,eps,eps,eps,eps]
+3:[eps,1,eps,eps,eps,eps]
+4:[eps,eps,1,eps,eps,eps]
+5:[eps,eps,eps,1,eps,eps]
+6:[eps,eps,eps,eps,1,eps]
+7:[eps,eps,eps,eps,eps,1]
+
+seed/8, seed%8
+*/
+
+vector<int> gen_val(Random& gen, int seed) {
+  seed /= 8;
+  int K = ((seed == 0 || seed == 1) ? 1000 : 1000000);
+  if (seed == 0 || seed == 2) {
+    int s = gen.uniform<int>(0, A_MAX - K + 1);
+    vector<int> A(K);
+    for (int i = 0; i < K; ++i) A[i] = i + s;
+    return A;
+  }
+  return gen.choice<int>(K, A_MIN, A_MAX);
+}
+
+vector<int> gen_prob(int seed) {
+  seed %= 8;
+  vector<int> S = {0, 1, 2, 3, 4, 5};
+  for (int t = 0; t < 6; ++t) {
+    bool add = 0;
+    if (seed == 0) add = 1;
+    if (seed == 1 && t != 1) add = 1;
+    if (seed == 2 && t == 0) add = 1;
+    if (seed == 3 && t == 1) add = 1;
+    if (seed == 4 && t == 1) add = 1;
+    if (seed == 5 && t == 1) add = 1;
+    if (seed == 6 && t == 1) add = 1;
+    if (seed == 7 && t == 1) add = 1;
```

Comment:
```suggestion
    if (seed == 0) add = 1;
    if (seed == 1 && t != 1) add = 1;
    if (seed == 2 && t == 0) add = 1;
    if (seed == 3 && t == 1) add = 1;
    if (seed == 4 && t == 2) add = 1;
    if (seed == 5 && t == 3) add = 1;
    if (seed == 6 && t == 4) add = 1;
    if (seed == 7 && t == 5) add = 1;
```

`gen_prob(seed)` ( 0 <= seed < 8 ) の頻度テーブルが現状以下のようになっています。これはジェネレータのコメントに書いてあることと異なります。

```
101 101 101 101 101 101 
101 1 101 101 101 101 
101 1 1 1 1 1 
1 101 1 1 1 1 
1 101 1 1 1 1 
1 101 1 1 1 1 
1 101 1 1 1 1 
1 101 1 1 1 1 
```